src/CustomerDataTable.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+import moment from "moment";
+
+
+class CustomerDataTable extends Component {
+
+constructor(props) {
+  super(props);
+  this.state = {
+    selected: false
+  }
+}
+
+
+toggleHighlightedRow = (e) => {
+  this.setState(prevState => ({
+    selected: !prevState.selected
+  })
+)
+this.state.selected ? (e.target.parentNode.className = 'bg-info') : (e.target.parentNode.className = '');
```

Comment:
This is something that you have to be careful about in React: you're directly manipulating the DOM. This is potentially a problem because now React doesn't know that your change is deliberate and will wipe away your change next time your render (anywhere in the app). Your manual DOM change is in the "real DOM", but not in React's virtual DOM and so it removes your manual change.

There's 2 approaches for fixing this problem:

1. We modify this approach to figure out which row needs to be highlighted (probably based on the row's id), and set that in state. Then when we're rendering we check if the current row needs to be highlighted or not
2. We separate our table into different row components that independently know whether they should be highlighted or not (hint: this approach is much simpler!). We looked at this way a little in class: remember how we could show multiple counters with different numbers for each one?

Reply:
Hey Ali! Thanks for this. Your commitments got me really keen to re-visit this and I've now added a revision with a new component. Let me know if this was along the right track.